### DISCUSSION TITLE
HTML and CSS

### DISCUSSION THREAD

**Discussion by big1oot** [microsoft/Web-Dev-For-Beginners#1309] [General]
Hello. I wrote an html code for a clone website, but I noticed that when executed with the browser, there is an issue. When the screen is reduced to a particular width, it begins to leave some elements out of the page. Can someone help me check what the issue is please?

**Comment by samueljaja00** [comment]
How do I help

**Reply by big1oot** [reply]
I don't know if it is possible for me to send you my code so you look through it and tell me where the problem lies. I'm kinda new to github, sorry.

**Reply by joanaquaicoe2-cmyk** [reply]
Okay I am too may be we should join a group to help us and our expenses

**Comment by bhavishyaPlawat** [comment]
> Hello. I wrote an html code for a clone website, but I noticed that when executed with the browser, there is an issue. When the screen is reduced to a particular width, it begins to leave some elements out of the page. Can someone help me check what the issue is please?

write a query pattern in css for the same , this happens coz you haven't told the website what to do when screen in small for example see my flipkart clone , the Nav bar gets changed when reduced to certain length mostly done for 3 screens ,laptop,tab and mobile

**Reply by big1oot** [reply]
I actually already used media query for the width, the query works, but there is still an issue i do not understand.

**Reply by bhavishyaPlawat** [reply]
maybe you are using pixelss frequenty for setting width and height ,try using vh,vw 
else upload the code let me see and try to fix, maybe i can help

**Reply by big1oot** [reply]
oh... uhm okay, i will check that, thank you.

**Reply by big1oot** [reply]
Do i just paste the code here, or how do i do that please? Thank you.

**Reply by bhavishyaPlawat** [reply]
make new repository on github
or search from youtube else paste the code i'll check

**Reply by joanaquaicoe2-cmyk** [reply]
Guys am just lost please help a friend please 🙏😊😊

**Comment by acquahinc** [comment]
Have you been able to sort it out?

**Reply by big1oot** [reply]
not yet please

**Comment by Akashgite1** [comment]
It sounds like you might be dealing with a responsive design issue. When elements are being cut off or not fitting properly within a reduced screen width, it’s often due to how CSS styles are being applied.

Here are a few common causes and solutions:

    Fixed Widths: If elements have a fixed width (e.g., width: 600px), they might not fit well on smaller screens. Consider using relative units like percentages (width: 100%) or flexible layouts (e.g., using flexbox or grid).

    Overflow: Check if any parent elements have overflow: hidden or overflow: auto, which might be clipping child elements. Ensure that your container elements can adapt to the size of their content.

    Media Queries: Make sure you’re using media queries to adjust styles for different screen sizes. For example:

@media (max-width: 768px) {
    .element {
        width: 100%;
        font-size: smaller;
    }
}

Box Model: Ensure padding and borders are properly accounted for. Sometimes, the total width of an element might exceed its container’s width due to padding or borders.

Viewport Meta Tag: If you’re working with responsive design, make sure you have the viewport meta tag in your HTML <head>:
html <meta name="viewport" content="width=device-width, initial-scale=1.0">

If you can provide more details or a snippet of your code, I might be able to offer more specific advice.

**Reply by big1oot** [reply]
omg, thank you so much, it worked. The issue was with the fixed width that i set. Thank you again. Have agreat day.

**Comment by Suryakanta311** [comment]
Use object-fit property in css and see what happen

**Comment by Suryakanta311** [comment]
> Hello. I wrote an html code for a clone website, but I noticed that when executed with the browser, there is an issue. When the screen is reduced to a particular width, it begins to leave some elements out of the page. Can someone help me check what the issue is please?

Bro uss vw insted of px

**Reply by joanaquaicoe2-cmyk** [reply]
at least you have done something what about me help me to drop something

**Comment by Nathan4508** [comment]
Hello, The problem is probably due to the lack of responsive design in your HTML and CSS code. @big1oot

**Comment by Suryakanta311** [comment]
Use vh and vw in size stuff

**Comment by bbelbuken** [comment]
be careful for fixed widths, absolute poisitioning. Instead use %, vw, rem, or flex/grid.

**Reply by FaraiAshley** [reply]
Hello my name is Ashley Masvosva on which page i can submit my assigments.l am new to this course

**Reply by FaraiAshley** [reply]
Hie my name is Ashley Masvosva.Where can i submit my assigments.l am new to this course

**Comment by HamdanOM** [comment]
Try Using Flexbox instead of fixed width and heights. Flexbox is a method to align items how you want without using fixed pixels for it. It may need some padding and margin, but that's it. It also makes it 10 times easier

**Comment by Pankajdas0025** [comment]
**It sounds like your layout isn’t fully responsive. This usually happens when some elements have fixed widths, no flex-wrap, or missing media queries.**

Try checking these things:

_Make sure your containers use width: 100% instead of fixed pixel widths._

_If you’re using flexbox, add flex-wrap: wrap; so items don’t overflow._

_Add media queries to adjust font sizes, images, and sections for smaller screens._

_Also check if any image or div is larger than the viewport — that often pushes the layout out._

**Reply by joanaquaicoe2-cmyk** [reply]
Hello please am new here please help me I want technology related stuff please help a friend I need I really need much help from people to help my experience

**Reply by joanaquaicoe2-cmyk** [reply]
Please thank you so much but please is there any way you can help me please 🤗 0553757458 Watt app please help me

**Comment by Pankajdas0025** [comment]
How I can help you

**Reply by joanaquaicoe2-cmyk** [reply]
Please I want to learn technology related stuff but I don't have much maybe you can start with the basics Thank you sooooooo much fam 🥺

**Comment by joanaquaicoe2-cmyk** [comment]
Thank you so much for your reply I so sorry I just saw this please is there
anyway you can help me

**Reply by bgantavya** [reply]
Hi @joanaquaicoe2-cmyk,
refer freecodecamp @youtube
or use my repo webdev it starts from basic html to building a fullstack app
